modin/core/dataframe/pandas/dataframe/dataframe.py

```diff
@@ -2432,56 +2435,53 @@ def _apply_func_to_range_partitioning(
                 # simply combine all partitions and apply the sorting to the whole dataframe
                 return self.combine_and_apply(func=func)
 
-        if is_numeric_dtype(self.dtypes[key_column]):
-            method = "linear"
+            if ideal_num_new_partitions < len(self._partitions):
```

Comment:
this logic was moved under the `if sampling_probability >= 1` condition so it would only take place when the `ideal_num_new_partitions` was actually modified